### PR TITLE
cloud: add AWS cracking automation

### PR DESCRIPTION
**Pull request by claudioandre-br**
The repository contains files for Terraform and Ansible in order to
assist users who need to run password cracking sessions in the Cloud.
*******
* How a real user will use the cluster?

### CODE REVIEW

**Comment by claudioandre-br** [comment]
Merging so that I can advertise and see if anyone is interested in the new functionality.

File: cloud-tool/playbook/install-john.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# the Free Software Foundation, as expressed in version 2, seen at
+# http://www.gnu.org/licenses/gpl-2.0.html
+######################################################################
+
+---
+- hosts: all
+  remote_user: ubuntu
+  tasks:
+    - name: Transfer the script
+      copy: src=../build-john.sh dest=/home/ubuntu/build-john.sh mode=0777
```
Comment (solardiz) [inline cloud-tool/playbook/install-john.yml:17]: Perhaps change 0777 to 0700 here.

File: cloud-tool/variables.tf [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+# http://www.gnu.org/licenses/gpl-2.0.html
+######################################################################
+
+# TODO ###############################################################
+# Change the cidr_blocks below to update using your IP address.
+variable ingress_data {
+  description = "The secutiry groups inbound rules."
+  type = map(object({description = string, cidr_blocks = list(string)}))
+  default = {
+    22 = { description = "Inbound SSH rule", cidr_blocks = [ "45.170.58.160/32" ] }
```
Comment (solardiz) [inline cloud-tool/variables.tf:18]: Hard-coded IP address here. (Also, typo in "secutiry" in at least two places.)

**Review by solardiz — COMMENTED** [review]
Like you, I don't know if this will be in demand, but I was curious to take a look at the diff here and noticed a few minor things.

Also, I suggest we standardize on "john" instead of "jtr" (or "JtR") in various component names, identifiers, etc.

**Comment by claudioandre-br** [comment]
It is useful for provisioning and destroying instances: it is fast, repeatable and error free. There is no possibility to forget to delete an instance or the EBS disk. It makes sense if someone wants to start a few instances and try a wordlist here, or a mask there, maybe incremental over there.
- you provision all instances at once;
- use SSH to perform your intended test on each instance;
- obtain the status of all instances at once;
- when you're done, destroy everything you've created locally using just one command.

File: cloud-tool/variables.tf (hunk 2 above)
Comment (claudioandre-br) [reply cloud-tool/variables.tf:18]: I can switch to "YOUR_IP_HERE/32". I can't think of another good numerical value.

**Pull request merged**